linetools/isgm/utils.py

```diff
@@ -284,70 +285,56 @@ def complist_from_table(table):
 
     """
     # Convert to QTable to handle units in individual entries more easily
-    table = QTable(table)
+    tkeys = table.keys()
 
     # mandatory and optional columns
-    min_columns = ['RA', 'DEC', 'ion_name', 'z_comp', 'vmin', 'vmax']
+    min_columns = ['RA', 'DEC', 'ion_name', 'z_comp', 'vmin', 'vmax', 'Z', 'ion', 'Ej']
     special_columns = ['name', 'comment', 'logN', 'sig_logN', 'flag_logN']
     for colname in min_columns:
         if colname not in table.keys():
             raise IOError('{} is a mandatory column. Please make sure your input table has it.'.format(colname))
 
-    #loop over rows
+    # Generate components -- Should add to linetools.isgm.utils
     complist = []
-    for row in table:
-        # mandatory
-        coord = SkyCoord(row['RA'].to('deg').value, row['DEC'].to('deg').value, unit='deg')  # RA y DEC must both come with units
-        Zion = name_to_ion(row['ion_name'])
-        zcomp = row['z_comp']
-        vlim =[row['vmin'].to('km/s').value, row['vmax'].to('km/s').value] * u.km / u.s  # units are expected here too
-
-        # special columns
-        try:
-            Ntuple = (row['flag_logN'], row['logN'], row['sig_logN'])  # no units expected
-        except KeyError:
+    coords = SkyCoord(ra=table['RA'], dec=table['DEC'], unit='deg')
```

Comment:
This could be dangerous, please check for the actual units in `RA` and `DEC`.

Reply:
What other units are there?  
Plus the Table may well not have units.

Reply:
Someone may try sexagesimal notation or whatever. I guess `SkyCoord` will give an error in that case, but I would prefer to raise an error if the units are not what we expect. Otherwise, what's the point on having units in the first place?

Reply:
We need to balance requiring units (which most people won't use)
with need.  I think this case is better my way.